numpy/lib/arraysetops.py

```diff
@@ -263,9 +263,9 @@ def _unique1d(ar, return_index=False, return_inverse=False,
         else:
             ret = (ar,)
             if return_index:
-                ret += (np.empty(0, np.bool),)
+                ret += (np.empty(0, np.intp),)
             if return_inverse:
-                ret += (np.empty(0, np.bool),)
+                ret += (np.empty(0, np.intp),)
             if return_counts:
                 ret += (np.empty(0, np.intp),)
         return ret
```

Comment:
you mean the else block where arr.size = 0? Your suggestion is definitely more elegant, and it looks like it will work, but I'm not sure it would be faster per se, since there would not be less instructions, unless I'm missing something here. I guess it is also easier to read in its current form (but more effort to maintain).